ProteoformSuiteInternal/ProteoformCommunity.cs

```diff
@@ -25,71 +22,89 @@ public bool has_e_and_t_proteoforms
-                        .ToList();
-            });
-
-            Parallel.ForEach(pfs1, pf1 => 
-            {
-                HashSet<string> pf1_prot_accessions = new HashSet<string>(pf1.candidate_relatives.OfType<TheoreticalProteoform>().Select(t => t.proteinList.FirstOrDefault().Accession + "_G" + t.proteinList.Count() + (t as TheoreticalProteoformGroup != null ? "_T" + ((TheoreticalProteoformGroup)t).accessionList.Count : "")));
-                foreach (string accession in pf1_prot_accessions)
                 {
-                    List<Proteoform> candidate_pfs2_with_accession = pf1.candidate_relatives.OfType<TheoreticalProteoform>().Where(t => t.proteinList.FirstOrDefault().Accession + "_G" + t.proteinList.Count() + (t as TheoreticalProteoformGroup != null ? "_T" + ((TheoreticalProteoformGroup)t).accessionList.Count : "") == accession).ToList<Proteoform>();
-                    candidate_pfs2_with_accession.Sort(Comparer<Proteoform>.Create((x, y) => Math.Abs(pf1.modified_mass - x.modified_mass).CompareTo(Math.Abs(pf1.modified_mass - y.modified_mass))));
-                    Proteoform best_pf2 = candidate_pfs2_with_accession.First();
-                    lock (best_pf2) lock (relations)
-                        relations.Add(new ProteoformRelation(pf1, best_pf2, relation_type, pf1.modified_mass - best_pf2.modified_mass));
-                }
-            });
+                    pf1.candidate_relatives = pfs2.Where(pf2 => allowed_relation(pf1, pf2, relation_type)).ToList();
 
-            return count_nearby_relations(relations.OrderBy(r => r.delta_mass).ToList());
-        }
+                    if (relation_type == ProteoformComparison.ExperimentalExperimental)
+                    {
+                        pf1.ptm_set = null;
+                        pf1.linked_proteoform_references = null;
+                        pf1.gene_name = null;
+                    }
 
-        
-        public List<ProteoformRelation> relate_ee(ExperimentalProteoform[] pfs1, ExperimentalProteoform[] pfs2, ProteoformComparison relation_type)
-        {
-            Parallel.ForEach(new HashSet<ExperimentalProteoform>(pfs1.Concat(pfs2)), pf =>
-            {
-                lock (pf) pf.candidate_relatives = pfs2.Where(pf2 => allowed_ee_relation(pf, pf2)).ToList<Proteoform>();
+                    if (relation_type == ProteoformComparison.ExperimentalTheoretical || relation_type == ProteoformComparison.ExperimentalDecoy)
+                    {
+                        ProteoformRelation best_relation = pf1.candidate_relatives
+                            .Select(pf2 => new ProteoformRelation(pf1, pf2, relation_type, pf1.modified_mass - pf2.modified_mass))
+                            .Where(r => r.candidate_ptmset != null) // don't consider unassignable relations for ET
+                            .OrderBy(r => r.candidate_ptmset.ptm_rank_sum) // get the best explanation for the experimental observation
+                            .FirstOrDefault();
+
```

Comment:
This is only taking one relation for each experimental, correct - the best one overall? Before, we would only allow one relation per gene, but for multiple genes there could be multiple relations